homeassistant/components/zamg/weather.py

```diff
@@ -67,12 +69,14 @@ async def async_setup_entry(
 class ZamgWeather(CoordinatorEntity, WeatherEntity):
     """Representation of a weather condition."""
 
+    _attr_attribution = ATTRIBUTION
+
     def __init__(
-        self, coordinator: ZamgDataUpdateCoordinator, name, station_id
+        self, coordinator: ZamgDataUpdateCoordinator, name: str, station_id: str
     ) -> None:
         """Initialise the platform with a data instance and station name."""
         super().__init__(coordinator)
-        self._attr_unique_id = f"{name}_{station_id}"
+        self._attr_unique_id = station_id
```

Comment:
We'll need a migration now for the entities where we've changed unique_id.

Example:
https://github.com/home-assistant/core/blob/83a6012f8061ef8c5c8d5f60cb0ded7282d0dd1e/homeassistant/components/bmw_connected_drive/__init__.py#L73-L107

Please also add tests that checks that the migration works.
https://github.com/home-assistant/core/blob/83a6012f8061ef8c5c8d5f60cb0ded7282d0dd1e/tests/components/bmw_connected_drive/test_init.py#L20-L74

Reply:
The station id is the same for all of the entities of the config entry. Just check if the unique_id is not the station id for the weather entity entries, and if so, update the entity entry to set the unique_id to the station id.